string_test.go

```diff
@@ -100,3 +100,150 @@ func TestRuneLength(t *testing.T) {
+		{"", "test123_string", "Test123String"},
+		{"", "test123string", "Test123String"},
+		{"", "", ""},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			actual := PascalCase(test.input)
+			if actual != test.expected {
+				t.Errorf("PascalCase(%q) = %q; expected %q", test.input, actual, test.expected)
+			}
+		})
+	}
+}
+
+func TestCamelCase(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    string
+		expected string
+	}{
+		{"", "hello_world", "helloWorld"},
+		{"", "helloWorld", "helloWorld"},
+		{"", "__hello_world-example string--", "helloWorldExampleString"},
+		{"", "WITH UPPERCASE LETTERS", "withUppercaseLetters"},
+		{"", "test123_string", "test123String"},
+		{"", "test123string", "test123String"},
+		{"", "", ""},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			result := CamelCase(test.input)
+			if result != test.expected {
+				t.Errorf("CamelCase(%q) = %q; want %q", test.input, result, test.expected)
+			}
+		})
+	}
+}
+
+func TestKebabCase(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    string
+		expected string
+	}{
+		{"", "hello world", "hello-world"},
+		{"", "HelloWorld", "hello-world"},
+		{"", "KebabCase", "kebab-case"},
+		{"", "already-kebab-case", "already-kebab-case"},
+		{"", "Already-Kebab-Case", "already-kebab-case"},
+		{"", "multiple   spaces", "multiple-spaces"},
+		{"", "", ""},
+		{"", "Single", "single"},
+		{"", "123_abs", "123-abs"},
+		{"", "SINGLE", "single"},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			result := KebabCase(test.input)
+			if result != test.expected {
+				t.Errorf("KebabCase(%q) = %q; want %q", test.input, result, test.expected)
+			}
+		})
+	}
+}
+
+func TestSnakeCase(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    string
+		expected string
+	}{
+		{"", "CamelCase", "camel_case"},
+		{"", "snakeCase", "snake_case"},
+		{"", "snake-case", "snake_case"},
+		{"", "SnakeCaseTest", "snake_case_test"},
+		{"", "Snake_Case_With_Underscores", "snake_case_with_underscores"},
+		{"", "lowercase", "lowercase"},
+		{"", "UPPERCASE", "uppercase"},
+		{"", "", ""},
+	}
+
+	for _, test := range tests {
+		t.Run(test.input, func(t *testing.T) {
+			got := SnakeCase(test.input)
+			if got != test.expected {
+				t.Errorf("SnakeCase(%q) = %q; want %q", test.input, got, test.expected)
+			}
+		})
+	}
+}
+
+func TestWords(t *testing.T) {
+	type args struct {
+		str string
+	}
+	tests := []struct {
+		name string
+		args args
+		want []string
+	}{
+		{"", args{"PascalCase"}, []string{"Pascal", "Case"}},
+		{"", args{"camelCase"}, []string{"camel", "Case"}},
+		{"", args{"snake_case"}, []string{"snake", "case"}},
+		{"", args{"kebab_case"}, []string{"kebab", "case"}},
+		{"", args{"_test text_"}, []string{"test", "text"}},
+		{"", args{"test123string"}, []string{"test", "123", "string"}},
+		{"", args{"UPPERCASE"}, []string{"UPPERCASE"}},
```

Comment:
This test should provide real life examples, and include things about Go initialisms

Maybe you could have to look at these repositories: 

https://github.com/client9/misspell

https://github.com/ettle/strcase

Reply:
I have updated the test cases